src/routes/DigitalAssets/routes/Send/sagas/digitalAssetsSend.js

```diff
@@ -516,27 +543,91 @@ function* setPriority(): Saga<void> {
     const digitalAsset: ExtractReturn<typeof selectDigitalAsset> =
       yield select(selectDigitalAsset, assetAddress)
 
-    if (!((digitalAsset &&
-          digitalAsset.blockchainParams &&
-          digitalAsset.blockchainParams.staticGasAmount)
-          || digitalAsset.isCustom)) {
-      // we can't call estimateGas for custom asset here, because we need a privateKey for it
+    if (!digitalAsset) {
       return
     }
 
+    const gasLimit = yield* requestGasLimit(digitalAsset)
+    if (gasLimit) {
+      yield put(digitalAssetsSend.setFormFieldValue('gasLimit', String(gasLimit)))
+    }
+  }
+}
+
+function* checkGasLimitWarning(newGasLimit: string): Saga<void> {
+  const {
+    formFieldWarnings,
+    initialGasSettings,
+  }: DigitalAssetsSendState = yield select(selectDigitalAssetsSend)
+
+  // Priority warnings
+  const gasLimit = parseInt(newGasLimit, 10)
+  const initialGasLimit = parseInt(initialGasSettings.gasLimit, 10)
+
+  if (!Number.isNaN(gasLimit) &&
+      !Number.isNaN(initialGasLimit) &&
+      gasLimit < initialGasLimit) {
     yield put(
-      digitalAssetsSend.setFormFieldValue('gasLimit', digitalAsset.blockchainParams.staticGasAmount)
+      digitalAssetsSend.setFormFieldWarning('gasLimit', 'Seems this transaction can fail')
     )
+  } else if (formFieldWarnings.gasLimit) {
+    yield put(digitalAssetsSend.setFormFieldWarning('gasLimit', ''))
+  }
+}
+
+function* checkGasPriceWarning(newGasPrice: void | string | BigNumber): Saga<void> {
+  const {
+    formFieldWarnings,
+    initialGasSettings,
+  }: DigitalAssetsSendState = yield select(selectDigitalAssetsSend)
+
+  // Priority warnings
+  try {
+    const gasPrice = toBigNumber(newGasPrice)
+    const initialGasPrice = toBigNumber(initialGasSettings.gasPrice)
+
+    if (gasPrice.lt(initialGasPrice.times(0.5)) || gasPrice.lt(21000)) {
```

Comment:
Better to set `21000` and `0.5` as named constants on top of the file. When we will return here with refactoring, we will know what do they mean.